### ISSUE TITLE
Xamarin Studio on Windows 8.1 and System.DllNotFoundException: CSteamworks

### ISSUE THREAD

**Issue by genail** [rlabrecque/Steamworks.NET#87]
Hi,

I'm using SteamWorks.net with Unity with no issues. Now I wanted to build an UGC uploader using Xamarin Studio, GTK# and SteamWorks.net on Windows. Here's what I did:

1. I've crated an empty GTK# project and set the runtime to Mono 4.2.2
2. It compiles and runs smoothly
3. I've copied Steamworks.NET directory including all the sources from 7.0.0 release.
4. I defined STEAMWORKS_WIN symbol
5. It still compiles
6. When run it shuts down with exception:

    Unhandled Exception:
    System.DllNotFoundException: CSteamworks
      at (wrapper managed-to-native) Steamworks.NativeMethods:SteamAPI_InitSafe ()
      at Steamworks.SteamAPI.Init () [0x00006] in     G:\Projects\TKoU\project_name\project_name\Steamworks.NET\Steam.cs:52 

7. I've copied CSteamworks.dll and steam_api.dll from Plugins/x86 to bin\Debug (this folder has the exe that I am trying to run)
8. The exception is still there

I searched over similar issues here like #23 and googled around, but still couldn't solve this issue. I ran the application with mono log level set to debug and found messages like this one:

Mono: DllImport error loading library 'G:\Projects\TKoU\project_name\project_name\bin\Debug\CSteamworks.dll': 'The system cannot find the file specified.

What is the most strange is that this file does exist and the path is OK (checked it from the command line). I also tried to inspect the library using Dependency Walker, but nothing seems to be off.

Thank you for any help!
Piotr

**Comment by rlabrecque** [comment]
Could you be trying to run a 64bit .NET application with the 32bit (x86) CSteamworks.dll/steam_api.dll?

**Comment by genail** [comment]
@rlabrecque Thanks for replying!

I did a double-check. In Project Options -> Build -> Compiler there's Platform target set to x86.
Is there any Xamarin example project available that I could use as a template?

**Comment by rlabrecque** [comment]
Have you tried using the precompiled DLL yet?

**Comment by genail** [comment]
@rlabrecque Yes, just did. Removed Steamworks.NET directory from the solution, added reference to Steamworks.NET.dll. It compiles fine, but I see the same error after trying to launch it.

P.S. To make sure that is using Steamworks.NET.dll I tried to compile the project without the DLL. It fails to find the namespace so I'm positive that I just did things correctly. Yet I cannot shake the feeling that I am doing something obviously wrong.

**Comment by genail** [comment]
@rlabrecque Well, I just solved it. I've just [changed](http://imgur.com/Yxo3goq) Active Runtime from Mono.4.2.2 to Microsoft.NET and it's running smoothly!

What is most surprising is that the project did not compile when I did that previously. Now it seems just fine. I do not recall changing anything from the first time I posted, but something has changed for sure and I don't know what.

Maybe that's because I [restarted](https://www.youtube.com/watch?v=nn2FB1P_Mn8) my Windows machine? Thank you for your help! :-)

Piotr
